libz-rs-sys/src/gz.rs

```diff
@@ -2269,6 +2269,64 @@ pub unsafe extern "C-unwind" fn gzgets(file: gzFile, buf: *mut c_char, len: c_in
     }
 }
 
+/// Dynamically update the compression level and strategy for `file`. See the
+/// description of [`deflateInit2`] for the meaning of these parameters. Previously
+/// provided data is flushed before applying the parameter changes.
+///
+/// # Returns
+///
+/// - `Z_OK` on success.
+/// - `Z_STREAM_ERROR` if the file was not opened for writing.
+/// - `Z_ERRNO` if there is an error writing the flushed data.
+/// - `Z_MEM_ERROR` if there is a memory allocation error.
+///
+/// # Safety
+///
+/// - `file`, if non-null, must be an open file handle obtained from [`gzopen`] or [`gzdopen`].
+#[cfg_attr(feature = "export-symbols", export_name = crate::prefix!(gzsetparams))]
+pub unsafe extern "C-unwind" fn gzsetparams(file: gzFile, level: c_int, strategy: c_int) -> c_int {
+    let Ok(strategy) = Strategy::try_from(strategy) else {
+        return Z_STREAM_ERROR;
+    };
```

Comment:
This check isn't in zlib-ng, but I needed it for the Rust version because we use a `Strategy` enum to hold the strategy internally.  I return `Z_STREAM_ERROR` here to for consistency with the Rust version of `deflateParams`.